next-cloudinary/src/components/UploadWidget/UploadWidget.js

```diff
@@ -0,0 +1,103 @@
+import { useRef } from "react";
+import Script from "next/script";
+
+const UploadWidget = ({ children, onUpload, options, signed }) => {
+  const cloudinary = useRef();
+  const widget = useRef();
+
+  /**
+   * generateSignature
+   * @description Makes a request to an endpoint to sign Cloudinary parameters as part of widget creation
+   */
+
+  function generateSignature(callback, paramsToSign) {
+    fetch(`/api/sign-cloudinary-params`, {
+      method: "POST",
+      body: JSON.stringify({
+        paramsToSign,
+      }),
+    })
+      .then((r) => r.json())
+      .then(({ signature }) => {
+        callback(signature);
+      });
+  }
+
+  /**
+   * createWidget
+   * @description Creates a new instance of the Cloudinary widget and stores in a ref
+   */
+
+  function createWidget() {
+    // When creating a signed upload, you need to provide both your Cloudinary API Key
+    // as well as a signature generator function that will sign any paramters
+    // either on page load or during the upload process. Read more about signed uploads at:
+    // https://cloudinary.com/documentation/upload_widget#signed_uploads
+
+    const totalOptions = {
+      ...options,
+      ...(!!signed && {
+        uploadSignature: generateSignature,
+      }),
+    };
+
+    if (signed && !totalOptions.apiKey) {
```

Comment:
should this be:
```
if (signed && ( !totalOptions.apiKey || !totalOptions.apiSecret ) ) {
```
?

Reply:
apiSecret is not needed as we only input apiKey ad secret is used in server side API instead